app/templates/user/profile.html

```diff
@@ -85,15 +85,21 @@ <h3 class="text-center">
                 {% endif %}
             {% else %}
                 <div class="flex-center-all"><h3 class="text-center">{{ user.username }}</h3></div>
-                {% if missing_wca_association %}
+                {% if missing_wca_association and viewing_self %}
                 <div class="flex-center-all">
                     <a href="{{ url_for('wca_assoc') }}" class="btn btn-lg btn-block btn-social btn-wca btn-force-btn-width">
                         <span><img class="wca-login-img" src="/static/images/wca_color.png" /></span>
                         Associate with WCA
                     </a>
                     <span style="font-weight: normal; padding-left: 5px;" class="fas fa-question-circle spacer" data-toggle="tooltip" data-placement="bottom" title="{{ wca_assoc_explanation }}"></span>
                 </div>
+                {% elif not missing_wca_association and (viewing_self or show_wca_id) %}
```

Comment:
You can check if the user's WCA id is set by just checking `user.wca_id` in a boolean context. In order to display the WCA ID for any user (not just viewing their own profile) I think this should be:

```python
{% elif (viewing_self or show_wca_id) and user.wca_id %}
...
```

This semantically translates to "if the user is viewing their own profile, or the user has "show WCA ID" setting turned on, *and* the user has a WCA ID.

Python can check non-booleans in boolean contexts (like we are doing above with `user.wca_id`). If the value is `None`, or is an empty string, it will evaluate as `False`. Otherwise it'll evaluate as `True`.